File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/crawler_runner/base/CouponUrlCrawlerBase.java
package com.huythanh0x.udemycoupons.crawler_runner.base;

import java.util.List;

/**
 * This class is an abstract base class for coupon url crawlers.
 * It contains a field storing the API URL used for crawling.
 * <p>
 * Child classes must implement the getAllCouponUrls method to retrieve a list of coupon URLs.
 */
public abstract class CouponUrlCrawlerBase {
    String apiUrl;

    /**
     * Returns a list of strings representing all coupon URLs.
     * Subclasses must implement this method to provide the actual implementation.
     *
     * @return a list of strings representing all coupon URLs
     */
    public abstract List<String> getAllCouponUrls();
}

File: modules/coupon-crawler-service/src/main/kotlin/com/thanh0x/coursedeal/crawlerrunner/base/CouponUrlCrawlerBase.kt
package com.thanh0x.coursedeal.crawlerrunner.base

/**
 * This class is an abstract base class for coupon url crawlers.
 * It contains a field storing the API URL used for crawling.
 *
 * Child classes must implement the getAllCouponUrls method to retrieve a list of coupon URLs.
 */
abstract class CouponUrlCrawlerBase {
    var apiUrl: String? = null

    /**
     * Returns a list of strings representing all coupon URLs.
     * Subclasses must implement this method to provide the actual implementation.
     *
     * @return a list of strings representing all coupon URLs
     */
    abstract fun getAllCouponUrls(): List<String>
}
